### ISSUE TITLE
Can't edit file; IndexOutOfBoundsException

### ISSUE THREAD

**Issue by haggholm** [carymrobbins/intellij-haskforce#190]
I can’t edit one file in my project at all—I can reposition the cursor, but not delete, backspace, enter new text, copy, cut…

Log: https://gist.github.com/haggholm/e9644a8a042b8c9c72be

The file is tiny, I had just started a project to learn (don’t really know Haskell). Probably not relevant, but just in case: https://gist.github.com/haggholm/2691c424d8ad792f63aa

**Comment by carymrobbins** [comment]
It seems that the parser isn't dealing with the missing `in` clause after the `let` binding.  Although, for it to affect the editor is terrible.  I've noticed issues like this before, and it seems like the parser shouldn't cause something like this.  It's either a bug in the parser that Grammar Kit generates, or possibly a bug in the `HaskellParserWrapper`.

In any case, you can probably get around this by completing the expression after the `let` with `in foo`.  I had no trouble adding that text to the end, so hopefully it should work for you - 
```haskell
    Int     -> let (y, m, d) = toGregorian(start) start : everyNth tp () in foo
```

I'll work on resolving this bug for the next release.

**Comment by haggholm** [comment]
The major problem is of course that I can’t edit the file due to the exception, so I’ll have to use another editor to fix any such mistakes as they happen—and since I’m just learning Haskell, and its syntax is rather opaque to a beginner, I expect to have a lot of syntax errors initially…

Ah well, I’ll live with another editor for now and look forward to the next release. Is there an ETA/roadmap?

**Comment by carymrobbins** [comment]
This shouldn't happen regularly.  Most syntax errors simply come up with proper error highlighting like you'd expect.  There just happens to be a bug somewhere in the parser that causes some weird issues in this specific case.  I was able to add text to the end of the file just fine, so I assumed it would work for you as well, but apparently that's not the case.

Long story short, you should still be able to use it in the meantime.  Sorry for the issues!  I'm hoping to get the next release out by the end of this week.

**Comment by carymrobbins** [comment]
Better yet, let's do a release now.  I can work around this issue for now by parsing incomplete `let` expressions, so that should handle this specific issue.  The fix will be available in **v0.3-beta.22**, which I am releasing here shortly.
